src/hera/workflows/workflow.py

```diff
@@ -153,99 +146,110 @@ def _build_templates(self) -> Optional[List[TTemplate]]:
                             self.volume_claim_templates.append(claim)
         return templates or None
 
+    mapped_model: ClassVar[Type[BaseModel]] = _ModelWorkflow
```

Comment:
As I pointed out in the PR desc. `ModelMapper` doesn't require subclassing, because i've added this field to each subclass, which tells the resource class itself to which model it refers.

Then I use `__init_subclass__` to perform the checks `ModelMapper` had been checking, at subclass creation time. This change could probably have happened as a separate PR. It was sort-of necessary to happen **for** this PR, but it could have happened independently of it, i think.